src/preset_cli/api/clients/superset.py

```diff
@@ -268,10 +269,19 @@ def _run_query(
         }
         self.session.headers.update(headers)
 
-        _logger.debug("POST %s\n%s", url, json.dumps(data, indent=4))
-        response = self.session.post(url, json=data)
-        validate_response(response)
+        try:
+            _logger.debug("POST %s\n%s", url, json.dumps(data, indent=4))
+            response = self.session.post(url, json=data)
+            response.raise_for_status()
 
+        # Legacy superset installations don't have the SQL API endpoint yet
+        except requests.exceptions.HTTPError as err:
+            if err.response.status_code == 404:
+                url = self.baseurl / "superset/sql_json/"
+                _logger.debug("POST %s\n%s", url, json.dumps(data, indent=4))
+                response = self.session.post(url, json=data)
```

Comment:
What happens if we have an error but it's not 404? We need to raise the exception if something other than a 404 happens. This can be simplified a little bit to do that:

```suggestion
        _logger.debug("POST %s\n%s", url, json.dumps(data, indent=4))
        response = self.session.post(url, json=data)
        
        # Legacy superset installations don't have the SQL API endpoint yet
        if response.status_code == 404:
            url = self.baseurl / "superset/sql_json/"
            _logger.debug("POST %s\n%s", url, json.dumps(data, indent=4))
            response = self.session.post(url, json=data)
            
        response.raise_for_status()
```

Reply:
that's totally right! Thanks for catching that!

Reply:
I don't think we need the `raise_for_status()` anymore so I've removed it as well.